### ISSUE TITLE
Chunks in `drasil-lang` that use `Maybe` instead of `Lenses`

### ISSUE THREAD

**Issue by Ant13731** [JacquesCarette/Drasil#2677]
Here is a list of types that use `Maybe` (referring to the [Rationalizing data structures project](https://github.com/JacquesCarette/Drasil/projects)). I will update these later to include the types/classes above and below each.

- `ConstrainedChunk` (`Maybe Expr`)
- `ConstrConcept` (`Maybe Expr`)
- `DefinedQuantityDict` (`Maybe UnitDefn`)
-  `IdeaDict` (`Maybe String`)
- `QuantityDict` (`Maybe Expr`)
- `MayHaveUnit` (`Maybe UnitDefn`)
- `HasDerivation` (lens to `Maybe Derivation`)
- `HasReasVal` (lens to `Maybe Expr`)
- `ListType`, `ListTuple`, `RawContent` (not a chunk, but still has `Maybe String` or `Maybe Int`)
- `Uncertainty` (`Maybe Double` and `Maybe Int`)

**Comment by balacij** [comment]
I believe that @Awurama-N has started work on the first 2 components (UncertainChunk + UncertQ). I'm not quite sure where the ticket went however.

**Comment by JacquesCarette** [comment]
Yes, @Awurama-N was working on that. I see that maybe there is an easier set of things to change -- the `Maybe UnitDefn` in `DefinedQuantityDict` should simply be deleted. Of course, it's not that easy! The proper process will involve seeing what the actual calls to the various `dqd` constructors are.  But this should be doable.

So @Awurama-N can you open a new issue, where the first part will be to see all calls to `dqd`, `dqdNoUnit`, `dqd'`, `dqdWr` and `dqdQd` in all of Drasil (not just the examples). We'll see what should be done from there.  This is definitely a really awful wart that can be fixed.

**Comment by balacij** [comment]
So, I believe this ticket was about analyzing which `Maybe` and trying to get rid of `Maybe` usage. However, I'm not sure I fully understand why we're trying to get rid of `Maybe`.

**Comment by samm82** [comment]
I think we may have backtracked on trying to do this and will continue to use `Maybe`. However, the specifics of when this happens throughout the Drasil process is yet to be determined and requires more thought

**Comment by JacquesCarette** [comment]
Right, I think I may have changed my mind on `Maybe`. We can and should use it for truly optional information.

We should never use it if we always expect to get a `Just` and getting a `Nothing` is an error.

**Comment by balacij** [comment]
I just skimmed through the results of `rg "error" -ths` and didn't see anything that looked like that. The only times we have a `fromMaybe` that default to an error on unwrapping involve things like `UID` referencing failing (which is not what this issue is about, and is fine).
